BASE/src/ServerTelemetryChannel/Managed/Shared/Implementation/NonWindowsIdentityProvider.cs

```diff
@@ -18,15 +18,25 @@ public NonWindowsIdentityProvider(IDictionary environment)
 
         string IIdentityProvider.GetName()
         {
+#if NETSTANDARD2_0
+            if (Environment.UserName != null)
+            {
+                return Environment.UserName;
+            }
+#endif
+
             // This variable is not guaranteed to be present. eg: in Docker containers.
             if (this.environment["USER"] != null)
             {
                 return this.environment["USER"].ToString();
-            }            
-            else
+            }
+
+            if (this.environment["USERNAME"] != null)
             {
-                return string.Empty;
+                return this.environment["USERNAME"].ToString();
```

Comment:
Can we use Environment.UserName all the time and not check for these environment variables ourselves?

Reply:
Environment.UserName should be avauilable in both net20 and net45